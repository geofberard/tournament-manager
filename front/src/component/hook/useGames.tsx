import useSWR from "swr";
import Game from "../../data/Game";
import { API_GAMES, fetchGames } from "../../service/GameService";

export const useGames: () => Game[] = () =>
    useSWR<Game[]>(API_GAMES, fetchGames, { suspense: true }).data;
