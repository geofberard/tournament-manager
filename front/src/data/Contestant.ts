import { Identified } from "./Identified";

export interface Contestant extends Identified{
    label: string;
}