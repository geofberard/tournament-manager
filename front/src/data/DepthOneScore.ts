import { Score } from "./Score";

export interface DepthOneScore extends Score {
    result: {[key:string]:Number};
}
