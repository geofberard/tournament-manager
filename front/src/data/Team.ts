import { Contestant } from "./Contestant";
import { Player } from "./Player";

export interface Team extends Contestant{
    name: string;
    players: Player[];
}
