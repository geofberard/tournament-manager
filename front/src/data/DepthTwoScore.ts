import { DepthOneScore } from "./DepthOneScore";
import { Score } from "./Score";

export interface DepthTwoScore extends Score {
    result: DepthOneScore[];
}
