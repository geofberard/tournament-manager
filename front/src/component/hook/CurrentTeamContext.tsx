import * as React from "react";
import {
    FC, PropsWithChildren, useContext, useEffect, useState,
} from "react";
import { Team } from "../../data/Team";
import * as TeamService from "../../service/TeamService";

interface CurrentTeamManager {
    currentTeam: Team;
    setCurrentTeam: (team: Team) => void;
}

const CurrentTeamContext: React.Context<CurrentTeamManager> = React.createContext(undefined);

export const CurrentTeamProvider: FC<PropsWithChildren> = ({ children }) => {
    const [currentTeam, setCurrentTeam] = useState<Team>();

    const setCurrentTeamAndPersist = (team: Team) => {
        setCurrentTeam(team);
        TeamService.setCurrentTeam(team);
    };

    useEffect(() => { 
        TeamService.getCurrentTeam()
        .then(setCurrentTeam)
        .catch(error => {
            console.log(error)
            setCurrentTeam(undefined)
        }) 
    }, []);

    return (
        <CurrentTeamContext.Provider value={{ currentTeam, setCurrentTeam: setCurrentTeamAndPersist }}>
            {children}
        </CurrentTeamContext.Provider>
    );
};

export const useCurrentTeam: () => [Team, (team: Team) => void] = () => {
    const manager = useContext(CurrentTeamContext);

    return [manager.currentTeam, manager.setCurrentTeam];
};
