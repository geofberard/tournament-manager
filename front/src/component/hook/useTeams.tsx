import * as React from "react";
import { useEffect, useState } from "react";
import { Team } from "../../data/Team";
import { API_TEAMS } from "../../service/TeamService";
import useSWR from "swr";
import { fetchJson } from "../../service/PromiseService";

export const useTeams: () => Team[] = () =>
    useSWR<Team[]>(API_TEAMS, fetchJson, { suspense: true }).data;
