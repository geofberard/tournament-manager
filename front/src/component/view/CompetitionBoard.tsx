import { Table, TableBody, TableCell, TableContainer, TableHead, TableRow, Badge, Grid } from "@mui/material";
import * as React from "react";
import { FC } from "react";
import { useCurrentTeam } from "../hook/CurrentTeamContext";
import { useTeams } from "../hook/useTeams";


export const CompetitionBoard: FC = () => {

    const teams = useTeams();
    const [currentTeam,] = useCurrentTeam();

    return (
        <Grid
            container
            item
            md={8}
            flexGrow={1}
            direction="column"
            justifyContent="center"
            alignItems="center"
        >
            <TableContainer>
                <Table>
                    <TableHead>
                        <TableRow>
                            <TableCell align="center">#</TableCell>
                            <TableCell align="center">Equipes</TableCell>
                            <TableCell align="center">P</TableCell>
                            <TableCell align="center">W</TableCell>
                            <TableCell align="center">L</TableCell>
                            <TableCell align="center">Pts</TableCell>
                        </TableRow>
                    </TableHead>
                    <TableBody>
                        {teams.map((team, index) => (
                            <TableRow
                            key={team.id}
                            sx={{ '&:last-child td, &:last-child th': { border: 0 } }}
                            selected={(team.id == currentTeam.id) ? true : false}
                            >
                                <TableCell align="center" component="th" scope="row">
                                    <Badge badgeContent={index + 1} ></Badge>
                                </TableCell>
                                <TableCell align="center">{team.name}</TableCell>
                                <TableCell align="center">0</TableCell>
                                <TableCell align="center">0</TableCell>
                                <TableCell align="center">0</TableCell>
                                <TableCell align="center">0 pts</TableCell>
                            </TableRow>
                        ))}
                    </TableBody>
                </Table>
            </TableContainer>
        </Grid>
    );
};