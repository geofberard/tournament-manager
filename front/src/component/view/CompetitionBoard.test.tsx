import * as React from "react";
import { mockFunctionReturn, testSnapshot } from "../../test/TestUtils";
import { CompetitionBoard } from "./CompetitionBoard";
import {useCurrentTeam} from "../hook/CurrentTeamContext";
import {useTeams} from "../hook/useTeams";
import {Team} from "../../data/Team";

const teamsMocked: Team[] = [
    { id: "team1", name: "Team1" },
    { id: "team2", name: "Team2" },
    { id: "team3", name: "Team3" },
];

jest.mock("../hook/CurrentTeamContext", () => ({
    useCurrentTeam: jest.fn(),
}));

jest.mock("../hook/useTeams", () => ({
    useTeams: jest.fn(),
}));

mockFunctionReturn(useTeams, teamsMocked);
mockFunctionReturn(useCurrentTeam, [teamsMocked[0], jest.fn()]);

testSnapshot(<CompetitionBoard />);
