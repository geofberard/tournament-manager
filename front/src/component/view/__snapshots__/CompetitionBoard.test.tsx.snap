// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match last snapshot 1`] = `
<ForwardRef(Grid)
  container={true}
  direction="column"
  flexGrow={1}
  item={true}
  md={7}
  sx={
    Object {
      "paddingLeft": Object {
        "md": "15px",
        "xs": "0",
      },
      "paddingRight": Object {
        "md": "15px",
        "xs": "0",
      },
    }
  }
>
  <ForwardRef(Grid)
    item={true}
    sx={
      Object {
        "borderBottom": [Function],
        "margin": "0 auto",
        "marginBottom": "5px",
        "maxWidth": "300px",
        "paddingBottom": Object {
          "md": "10px",
          "xs": "0",
        },
        "paddingTop": "10px",
      }
    }
  >
    <ForwardRef(Typography)
      variant="h3"
    >
      Classement
    </ForwardRef(Typography)>
  </ForwardRef(Grid)>
  <ForwardRef(TableContainer)
    style={
      Object {
        "borderRadius": "15px",
      }
    }
  >
    <ForwardRef(Table)>
      <ForwardRef(TableHead)>
        <ForwardRef(TableRow)>
          <Styled(ForwardRef(TableCell))
            align="center"
          >
            #
          </Styled(ForwardRef(TableCell))>
          <Styled(ForwardRef(TableCell))
            align="center"
          >
            Equipes
          </Styled(ForwardRef(TableCell))>
          <Styled(ForwardRef(TableCell))
            align="center"
          >
            P
          </Styled(ForwardRef(TableCell))>
          <Styled(ForwardRef(TableCell))
            align="center"
          >
            W
          </Styled(ForwardRef(TableCell))>
          <Styled(ForwardRef(TableCell))
            align="center"
          >
            L
          </Styled(ForwardRef(TableCell))>
          <Styled(ForwardRef(TableCell))
            align="center"
          >
            Pts
          </Styled(ForwardRef(TableCell))>
        </ForwardRef(TableRow)>
      </ForwardRef(TableHead)>
      <ForwardRef(TableBody)>
        <ForwardRef(TableRow)
          selected={true}
          sx={
            Object {
              "&:last-child td, &:last-child th": Object {
                "border": 0,
              },
            }
          }
        >
          <ForwardRef(TableCell)
            align="center"
            component="th"
            scope="row"
          >
            <ForwardRef(Badge)
              badgeContent={1}
            />
          </ForwardRef(TableCell)>
          <ForwardRef(TableCell)
            align="center"
          >
            Team1
          </ForwardRef(TableCell)>
          <ForwardRef(TableCell)
            align="center"
          >
            0
          </ForwardRef(TableCell)>
          <ForwardRef(TableCell)
            align="center"
          >
            0
          </ForwardRef(TableCell)>
          <ForwardRef(TableCell)
            align="center"
          >
            0
          </ForwardRef(TableCell)>
          <ForwardRef(TableCell)
            align="center"
          >
            0 pts
          </ForwardRef(TableCell)>
        </ForwardRef(TableRow)>
        <ForwardRef(TableRow)
          selected={false}
          sx={
            Object {
              "&:last-child td, &:last-child th": Object {
                "border": 0,
              },
            }
          }
        >
          <ForwardRef(TableCell)
            align="center"
            component="th"
            scope="row"
          >
            <ForwardRef(Badge)
              badgeContent={2}
            />
          </ForwardRef(TableCell)>
          <ForwardRef(TableCell)
            align="center"
          >
            Team2
          </ForwardRef(TableCell)>
          <ForwardRef(TableCell)
            align="center"
          >
            0
          </ForwardRef(TableCell)>
          <ForwardRef(TableCell)
            align="center"
          >
            0
          </ForwardRef(TableCell)>
          <ForwardRef(TableCell)
            align="center"
          >
            0
          </ForwardRef(TableCell)>
          <ForwardRef(TableCell)
            align="center"
          >
            0 pts
          </ForwardRef(TableCell)>
        </ForwardRef(TableRow)>
        <ForwardRef(TableRow)
          selected={false}
          sx={
            Object {
              "&:last-child td, &:last-child th": Object {
                "border": 0,
              },
            }
          }
        >
          <ForwardRef(TableCell)
            align="center"
            component="th"
            scope="row"
          >
            <ForwardRef(Badge)
              badgeContent={3}
            />
          </ForwardRef(TableCell)>
          <ForwardRef(TableCell)
            align="center"
          >
            Team3
          </ForwardRef(TableCell)>
          <ForwardRef(TableCell)
            align="center"
          >
            0
          </ForwardRef(TableCell)>
          <ForwardRef(TableCell)
            align="center"
          >
            0
          </ForwardRef(TableCell)>
          <ForwardRef(TableCell)
            align="center"
          >
            0
          </ForwardRef(TableCell)>
          <ForwardRef(TableCell)
            align="center"
          >
            0 pts
          </ForwardRef(TableCell)>
        </ForwardRef(TableRow)>
      </ForwardRef(TableBody)>
    </ForwardRef(Table)>
  </ForwardRef(TableContainer)>
</ForwardRef(Grid)>
`;
